libvast/src/system/partition.cpp

```diff
@@ -537,6 +543,7 @@ active_partition_actor::behavior_type active_partition(
             [=](caf::error err) {
               VAST_ERROR(self, "failed to persist indexer for", kv.first.fqn(),
                          "with error:", render(err));
+              ++self->state.persisted_indexers;
```

Comment:
Hm, this is inconsistent now: Either we ignore single failed indexers and just write as much as we can (current behavior), then we shouldnt deliver the promise here. Or we fail hard, then we shouldnt increase `persisted_indexers`, otherwise we might try to deliver it later again if the failed indexer was not the last one that arrived.

Reply:
I opted for the fail hard approach for now.